### ISSUE TITLE
Support for non-utf8/binary-safe data

### ISSUE THREAD

**Issue by prashanthpai** [RedisLabsModules/redismodule-rs#134]
The C API for command implementation is as follows:
```c
int Some_RedisCommand(RedisModuleCtx *ctx, RedisModuleString **argv, int argc) {
```
where `RedisModuleString` objects are binary safe C buffers.

However, the Rust's implementation is:
```rust
fn some_cmd(ctx: &Context, args: Vec<String>) -> RedisResult {
```
where args is a vector of `String`s which can only contain valid UTF-8 bytes.

Further, most APIs implemented in redismodule-rs accept and return either `String` or `&str`.

How about changing the command implementation signature to accept `Vec<RedisString>` rust structure which abstracts access to the C structure`RedisModuleString` ?

**Comment by slavak** [comment]
I believe this was done mostly for convenience of use. However in the case of command implementations, accepting a `Vec<RedisString>` might be acceptable ergonomically, as the `NextArg` trait can still be used to decode strings where needed.

This would be a backwards-compatibility breaking API change, though.

**Comment by gavrie** [comment]
I agree; it would be nice to be able to support non-UTF-8 data. I'm not worried about breaking the API; we're still in an early stage with this project and it's fine.
@prashanthpai if you'd like to attempt this, that'd be great.

**Comment by gkorland** [comment]
@prashanthpai please notice since v0.20.0 we moved to RedisString that `supports` none utf8 values
https://github.com/RedisLabsModules/redismodule-rs/releases/tag/v0.20.0

**Comment by markmcdowell** [comment]
@gkorland I notice the comment https://github.com/RedisLabsModules/redismodule-rs/blob/39f757feb662361dd0e7831a4699ca5c01ef2917/src/redismodule.rs#L192 which I assume means it doesn't quite support non-utf8 values yet.

All the API calls still take either `&str` or `String` as well.

**Comment by ss025** [comment]
Is there any update on this? I ran into the same issue while retrieving non-utf8 data. Or is there any alternative to store non-utf8 data

` CallReply::String(reply) => RedisValue::SimpleString(reply.to_string().unwrap()),`

Error
```
thread '<unnamed>' panicked at cargo/registry/src/index.crates.io-6f17d22bba15001f/redis-module-2.0.7/src/redisvalue.rs:236:84:
called `Option::unwrap()` on a `None` value
```
